cache/network/download.go

```diff
@@ -45,31 +47,18 @@ func Download(params DownloadParams, logger log.Logger) (matchedKey string, err
 	}
 
 	logger.Debugf("Download archive")
-	file, err := os.Create(params.DownloadPath)
-	if err != nil {
-		return "", fmt.Errorf("can't open download location: %w", err)
-	}
-	defer func(file *os.File) {
-		err := file.Close()
-		if err != nil {
-			logger.Warnf(err.Error())
-		}
-	}(file)
 
-	respBody, err := client.downloadArchive(restoreResponse.URL)
-	if err != nil {
-		return "", fmt.Errorf("failed to download archive: %w", err)
-	}
-	defer func(respBody io.ReadCloser) {
-		err := respBody.Close()
-		if err != nil {
-			logger.Warnf(err.Error())
-		}
-	}(respBody)
-	_, err = io.Copy(file, respBody)
-	if err != nil {
-		return "", fmt.Errorf("failed to save archive to disk: %w", err)
+	downloadErr := downloadFile(ctx, retryableHTTPClient.StandardClient(), restoreResponse.URL, params.DownloadPath)
```

Comment:
hmm.. in all other places in the api we use DefaultClient, i.e. this way:
retryableHTTPClient.Do(), which is hashicorp's cleanhttpclient with some default retries/backoffs

here we are passing StandardClient, which is Go's default client

is this intentional?

Reply:
i assume the lib expects standard lib Go's client

Reply:
yes, it requires a standard library *http.Client. `StandardClient()` gives back a wrapped *http.Client that is using the retry logic. So, the downloader is using a retry client under the hood to download each range. We tested that if one of the ranges download fails the client will retry it.